compiler/transforms/set_memory_layout.py

```diff
@@ -127,41 +120,29 @@ class AddMemoryLayout(RewritePattern):
 
     @op_type_rewrite_pattern
     def match_and_rewrite(
-        self, op: linalg.GenericOp | stream.StreamingRegionOp, rewriter: PatternRewriter
+        self, op: stream.StreamingRegionOp, rewriter: PatternRewriter
     ):
         # check if operation is dispatched via library call, as set by e.g.
         # the dispatch-kernels pass
 
-        if isinstance(op, linalg.GenericOp):
-            if op.library_call is None:
-                return
-            else:
-                library_call = op.library_call.data
-        elif isinstance(op, stream.StreamingRegionOp):
-            if op.accelerator is None:
-                return
-            else:
-                library_call = op.accelerator.data
+        if op.accelerator is None:
+            return
+        else:
+            library_call = op.accelerator.data
 
         has_add_c = False
 
         # check for library call
         if library_call == "snax_gemmx" or library_call == "snax_gemmx_stream":
             # only do so for qmac kernels
-            if isinstance(op, linalg.GenericOp):
-                if not isinstance(op.body.block.first_op, QMacOp):
-                    return
-            elif isinstance(op, stream.StreamingRegionOp):
-                assert isinstance(
-                    generic_op := op.body.block.first_op, stream.GenericOp
-                )
-                if not isinstance(generic_op.body.block.first_op, QMacOp):
-                    return
+            assert isinstance(generic_op := op.body.block.first_op, stream.GenericOp)
```

Comment:
nit: Feels a bit weird to assign a variable inside an assertion?
```suggestion
            generic_op = op.body.block.first_op
            assert isinstance(generic_op, stream.GenericOp)
```